jpsShared/src/org/jetbrains/jps/incremental/scala/model/HydraSettingsImpl.java

```diff
@@ -42,5 +52,9 @@ public void applyChanges(@NotNull HydraSettingsImpl hydraSettings) {
   public static class State {
     public boolean isHydraEnabled = false;
     public String hydraVersion = "";
+    public String noOfCores = Integer.toString((int) Math.ceil(Runtime.getRuntime().availableProcessors()/2D));
+    public String hydraStorePath = "";
+    public String sourcePartitioner = "";
+    public String projectRoot = "";
```

Comment:
Can we have these as Path or because of serialization/deserialization we need to keep using String? (wonder what they do in the Scala support)

Reply:
We need to keep this as String because of serialization/deserialization